base_report_to_printer/models/printing_printer.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (C) 2011 Agile Business Group sagl (<http://www.agilebg.com>)
+# Copyright (C) 2011 Domsense srl (<http://www.domsense.com>)
+# Copyright (C) 2013-2014 Camptocamp (<http://www.camptocamp.com>)
+# Copyright (C) 2016 SYLEAM (<http://www.syleam.fr>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+import os
+from tempfile import mkstemp
+
+from odoo import models, fields, api
+
+
+_logger = logging.getLogger(__name__)
+
+
+class PrintingPrinter(models.Model):
+    """
+    Printers
+    """
+
+    _name = 'printing.printer'
+    _description = 'Printer'
+    _order = 'name'
+
+    name = fields.Char(required=True, index=True)
+    server_id = fields.Many2one(
+        comodel_name='printing.server', string='Server', required=True,
+        help='Server used to access this printer.')
+    job_ids = fields.One2many(
+        comodel_name='printing.job', inverse_name='printer_id', string='Jobs',
+        help='Jobs printed on this printer.')
+    system_name = fields.Char(required=True, index=True)
+    default = fields.Boolean(readonly=True)
+    status = fields.Selection(
+        selection=[
+            ('unavailable', 'Unavailable'),
+            ('printing', 'Printing'),
+            ('unknown', 'Unknown'),
+            ('available', 'Available'),
+            ('error', 'Error'),
+            ('server-error', 'Server Error'),
+        ],
+        required=True,
+        readonly=True,
+        default='unknown')
+    status_message = fields.Char(readonly=True)
+    model = fields.Char(readonly=True)
+    location = fields.Char(readonly=True)
+    uri = fields.Char(string='URI', readonly=True)
+
+    @api.multi
+    def _prepare_update_from_cups(self, cups_connection, cups_printer):
+        mapping = {
+            3: 'available',
+            4: 'printing',
+            5: 'error'
+        }
+        vals = {
+            'name': cups_printer['printer-info'],
+            'model': cups_printer.get('printer-make-and-model', False),
+            'location': cups_printer.get('printer-location', False),
+            'uri': cups_printer.get('device-uri', False),
+            'status': mapping.get(cups_printer.get(
+                'printer-state'), 'unknown'),
+            'status_message': cups_printer.get('printer-state-message', ''),
+        }
+        return vals
+
+    @api.multi
+    def print_options(self, report=None, format=None, copies=1):
+        """ Hook to set print options """
+        options = {}
+        if format == 'raw':
+            options['raw'] = 'True'
+        if copies > 1:
+            options['copies'] = str(copies)
+        return options
+
+    @api.multi
+    def print_document(self, report, content, format, copies=1):
+        """ Print a file
+
+        Format could be pdf, qweb-pdf, raw, ...
+
+        """
+        self.ensure_one()
+        fd, file_name = mkstemp()
+        try:
+            os.write(fd, content)
```

Comment:
Changing this to content[0] resolved traceback

Reply:
Thanks.
This is because the `render_qweb_pdf` method returns a tuple containing the document (contents) and the format ('pdf' here).
I'll fix it in my next commit ;)